aspnetcore/razor-pages/razor-pages-conventions.md

```diff
@@ -194,6 +194,44 @@ Request the sample's About page at `localhost:5000/About/GlobalRouteValue/AboutR
 
 ![About page is requested with route segments for GlobalRouteValue and AboutRouteValue. The rendered page shows that the route data values are captured in the OnGet method of the page.](razor-pages-conventions/_static/about-page-global-and-about-templates.png)
 
+::: moniker range=">= aspnetcore-2.2"
+
+**Use a parameter transformer to customize page routes**
+
+Page routes generated by ASP.NET can be customized using a parameter transformer. A parameter transformer implements `IOutboundParameterTransformer` and transforms the value of parameters. For example, a custom `SlugifyParameterTransformer` parameter transformer changes the `SubscriptionManagement` route value to `subscription-management`.
+
+The `PageRouteTransformerConvention` is an page route model convention that applies a parameter transformer to the folder and file name segments of automatically generated page routes in an application. For example, the Razor page file at `/Pages/SubscriptionManagement/ViewAll.cshtml` would have its route rewritten from `/SubscriptionManagement/ViewAll` to `/subscription-management/view-all`.
+
+`PageRouteTransformerConvention` only transforms the automatically generated segments of a page route that come from the Razor page folder and file name. It doesn't transform route segments added with the `@page` directive. The convention also doesn't transform routes added by [AddPageRoute](/dotnet/api/microsoft.extensions.dependencyinjection.pageconventioncollectionextensions.addpageroute).
+
+The `PageRouteTransformerConvention` is registered as an option in `ConfigureServices`.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddMvc()
+        .AddRazorPagesOptions(options =>
+            {
+                options.Conventions.Add(
+                    new PageRouteTransformerConvention(
+                        new SlugifyParameterTransformer()));
+            });
+}
+
+public class SlugifyParameterTransformer : IOutboundParameterTransformer
+{
+    public string TransformOutbound(object value)
+    {
+        if (value == null) { return null; }
+
+        // Slugify value
+        return Regex.Replace(value.ToString(), "([a-z])([A-Z])", "$1-$2").ToLower();
+    }
+}
+```
+
```

Comment:
Should this link to the section about parameter transformers in general, or for MVC?

Reply:
There isn't a place that talks about parameter transformers in general. The fundamentals page talks about them as a reference in a route pattern. Leave this as is IMO.